tests/PregTest.php

```diff
@@ -201,4 +202,26 @@ public function testCorrectnessForNonUtf8String()
         static::assertSame([\chr(224)], $methodMatches);
         static::assertNotSame([\chr(224)], $functionMatches);
     }
+
+    /**
+     * Preg class is final, to test these internals reflection is necessary.
```

Comment:
don't test internals.
it's bad practice.

Reply:
Sure. Now the internals are publicly tested. Equally bad practice, but not that visible. Also a benefit of more invalid pattern examples. See as well your other comment about the fixtures.